src/Services/BasketService.php

```diff
@@ -462,8 +440,10 @@ private function parseBasketItemOrderParams(array $basketOrderParams): array
      */
     public function updateBasketItem(int $basketItemId, array $data): array
     {
+        $basket = $this->getBasket();
         $data['id'] = $basketItemId;
         try {
+            $this->couponService->validateBasketItemUpdate($basket, $data, $this->getBasketItem($basketItemId));
```

Comment:
erst die Variable $basketItem füllen und dann übergeben, kein funktionsaufruf als parameter wie in Zeile 489